### ISSUE TITLE
What is the impact of "at_least" option?

### ISSUE THREAD

**Issue by andreyb051** [fvutils/pyvsc#71]
Hi,

I need to get at least x entries in each bin, before declaring coverage as 100%, but I got coverage at 100% with a single entry in each bin. "at_least" option had no effect on coverage. So how to do sampling, as long as there are less than x entries in each bin?

Thanks,
  Andrey

**Comment by edcote** [comment]
This seems to be a question more than an issue, please close and move to:

https://github.com/fvutils/pyvsc/discussions

**Comment by mballance** [comment]
Hi @andreyb051,
    Thanks for raising this issue. I created a small testcase, and can see that something isn't quite as expected.

```
        @vsc.covergroup
        class cg(object):
            
            def __init__(self):
                self.with_sample(dict(
                    a=vsc.uint8_t(),
                    b=vsc.uint8_t()))
                
                self.options.at_least = 2
                
                self.cp1 = vsc.coverpoint(self.a, bins={
                    "a" : vsc.bin_array([], 1, 2, 4, 8),
                    })
                self.cp2 = vsc.coverpoint(self.b, bins={
                    "b" : vsc.bin_array([], 1, 2, 4, 8)
                    })
                
        cg_i = cg()
        
        cg_i.sample(1, 1)
        cg_i.sample(1, 2)
        cg_i.sample(2, 4)
        cg_i.sample(2, 8)
        
        vsc.report_coverage(details=True)
```

The bin hits are as expected, however the total coverage achieved appears to assume an 'at_least' of 1.
```
TYPE cg : 75.000000%
    CVP cp1 : 50.000000%
    Bins:
        a[0] : 2
        a[1] : 2
        a[2] : 0
        a[3] : 0
    CVP cp2 : 100.000000%
    Bins:
        b[0] : 1
        b[1] : 1
        b[2] : 1
        b[3] : 1
    INST cg : 75.000000%
        CVP cp1 : 50.000000%
        Bins:
            a[0] : 2
            a[1] : 2
            a[2] : 0
            a[3] : 0
        CVP cp2 : 100.000000%
        Bins:
            b[0] : 1
            b[1] : 1
            b[2] : 1
            b[3] : 1
```

Unfortunately, this doesn't appear to be a 5min investigation, so I'll have to defer it to the weekend. Definitely appears to be a legit issue, though.

-Matthew

**Comment by mballance** [comment]
Hi @andreyb051,
    Okay, I've been able to dig into this more deeply and things are working better. Instance bins now properly respect 'at_least' specified on their respective covergroups/coverpoints/crosses. Options handling is a bit of a sizable area, and I suspect there are more bugs hiding here, but this is an improvement at least.

The fixes for this issue are in the 0.3.5 release of PyVSC. This is now available on PyPi and GitHub. Do note that you'll also need to update the PyUCIS library either from PyPi or GitHub.

Thanks again for raising this!

-Matthew
